editor/src/components/canvas/ui/floating-insert-menu.tsx

```diff
@@ -647,11 +617,10 @@ export var FloatingMenu = React.memo(() => {
         const pickedInsertableComponent = (value as InsertMenuItem).value
 
         function getActionsToDispatch() {
-          switch (pickedInsertableComponent.element) {
-            case 'conditional':
-              return onChangeConditional()
-            case 'fragment':
-              return onChangeFragment()
+          switch (pickedInsertableComponent.element.type) {
+            case 'JSX_CONDITIONAL_EXPRESSION':
+            case 'JSX_FRAGMENT':
+              return onChangeConditionalOrFragment(pickedInsertableComponent.element)
             default:
```

Comment:
I think at this point you could make it explicitly handle JSX_ELEMENT and include a guard at the end!

Reply:
https://github.com/concrete-utopia/utopia/pull/3466/commits/dbb522dc20ee009c9dccff1a7cd82bc6609afce2